### PR TITLE
Redirected full upgrade logs to upgrade-diff-logs dir

### LINKED ISSUE #479: LogAnalyzer: Redirect full logs to a directory with separate log files
Redirect the full Logs to a Directory having log files for every log file of system.

### PR DESCRIPTION
**Pull request by renzon**
close #479

### CODE REVIEW

File: automation_tools/satellite6/log.py [hunk 1]
```diff
@@ -105,7 +106,24 @@ def get_line_count():
         execute(get_line_count, host=self.host)
 
 
-def analyze(log_file, content):
+def _save_full_log(host, log_file_path, content):
+    """Save full log on upgrade-diff-logs dir
+
+    :param log_file_path: file name path
```
Comment (jyejare) [inline automation_tools/satellite6/log.py]: Missing docstring for host parameter

File: automation_tools/satellite6/log.py [hunk 2]
```diff
@@ -105,7 +106,24 @@ def get_line_count():
...
+        os.mkdir(dir_path)
+    log_file_name = os.path.split(log_file_path)[-1]
+    log_file_name = '%s-%s' % (host, log_file_name)
+    file_path = os.path.join(dir_path, log_file_name)
+    with open(file_path, 'wb') as log_file:
+        log_file.write(content)
+        _print_wrapper('## Full upgrade logs saved on %s' % file_path)
+
+
+def analyze(host, log_file, content):
```
Comment (jyejare) [inline automation_tools/satellite6/log.py]: Missing docstring of host parameter

File: automation_tools/satellite6/log.py [hunk 3]
```diff
@@ -115,8 +133,12 @@ def analyze(log_file, content):
 
     def print_lines(lines_enumeration):
         """print lines with numbers"""
+        no_line_flag = True
         for i, line in lines_enumeration:
             _print_wrapper('{}: {}'.format(i, line))
+            no_line_flag = False
+        if no_line_flag:
+            _print_wrapper('None')
```
Comment (jyejare) [inline automation_tools/satellite6/log.py]: I this is not required to print 'None'. We can leave as it is or we can return 'False' and operate accordingly.

File: tests/test_log.py [hunk 4]
```diff
@@ -173,13 +179,7 @@ def print_mock(mocker):
...
 REPORT_WITHOUT_ERROR = '''### Analyzing /var/log/candlepin/candlepin.log:
 ## Errors found:
-## Full log:
-1: 2016-11-23 05:41:54 [app] [I] Processing by Api::
-2: 2016-11-23 05:41:54 [app] [I] Parameters: {"report"=>"[FILTERED]", "apiv"=>
-3: 2016-11-23 05:41:54 [app] [I] processing report for
-4: 2016-11-23 05:41:54 [app] [I] Imported report for
-5: 2016-11-23 05:41:55 [app] [I] Rendered api/v2/reports/create.json (5.5ms)
-6: 2016-11-23 05:41:55 [app] [I] Completed 201 Created in 61ms'''
+None'''
```
Comment (jyejare) [inline tests/test_log.py]: We can remove this, if we are not printing 'None'

File: tests/test_log.py [hunk 5]
```diff
@@ -218,33 +218,13 @@ def print_mock(mocker):
...
-17: undefined method `import_data' for nil:NilClass (NoMethodError)
 18: undefined method `[]' for nil:NilClass (NoMethodError)'''
 
 CONTENT_WITH_SPECIAL_CHAR = 'éçã'.encode('utf-8')
 
 REPORT_WITH_SPECIAL_CHAR = '''### Analyzing /var/log/candlepin/candlepin.log:
 ## Errors found:
-## Full log:
-1: éçã'''
+None'''
```
Comment (jyejare) [inline tests/test_log.py]: Same as above

Comment (renzon) [inline tests/test_log.py]: I was intending to give some kind of feedback when no error is found. So you really prefer not having it?

Comment (jyejare) [inline tests/test_log.py]: You can:
Either, remove None and Dont print anything
OR, Print some feedback on not finding anything. 

I prefer 2nd option which you are also ok with.

**Comment by renzon** [comment]
@jyejare fix done.

**Comment by renzon** [comment]
For the one who will merge this: please use squash and merge

**Comment by sghai** [comment]
ACK

**Pull request merged**